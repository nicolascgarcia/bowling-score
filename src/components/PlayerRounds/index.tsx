import { RefObject } from 'react';
import { Player } from '../../contexts/GameContext/gameContext.types';
import Container from '../Container';
import ScoreColumn from '../ScoreColumn';
import UserName from '../UserName';

type PlayerRoundsProps = {
	user: Player;
	userIndex: number;
	playerRef: RefObject<HTMLDivElement>;
	currentPlayer: number;
};

export default function PlayerRounds({
	user,
	userIndex,
	playerRef,
	currentPlayer,
}: PlayerRoundsProps) {
	return (
		<Container
			ref={playerRef}
			minWidth='1500px'
			bg='#121214'
			borderWidth={currentPlayer === userIndex ? '3px' : '0px'}
			borderStyle='solid'
			borderColor='#8257e5'
			display='flex'
			flexDirection='column'
			mb={4}
			p={3}
			borderRadius='10px'
		>
			<Container width={1} display='flex' alignItems='center'>
				<img src={user.photo} alt={user.name} width={50} height={50} />
				<Container ml={3}>
					<UserName>{user.name}</UserName>
				</Container>
			</Container>
			<Container
				display='flex'
				borderColor='gray'
				borderWidth='2px'
				borderStyle='solid'
				borderRadius='10px'
			>
				{user.scores.map((score, scoreIndex) => (
					<ScoreColumn
						key={`${Math.random() * scoreIndex}`}
						currentRound={user.currentRound}
						scoreIndex={scoreIndex}
						currentPlayer={currentPlayer}
						userIndex={userIndex}
						scoreType={score.type}
						firstShot={score.firstShot}
						secondShot={score.secondShot}
						thirdShot={score.thirdShot}
						score={score.score}
					/>
				))}
				<Container
					display='flex'
					flex={1}
					flexDirection='column'
					borderRightColor='gray'
					borderRightWidth='2px'
					borderRightStyle='solid'
				>
					<Container
						width={1}
						borderBottomColor='gray'
						borderBottomWidth='2px'
						borderBottomStyle='solid'
					>
						<h1
							style={{
								textAlign: 'center',
								fontWeight: 'bold',
								color: '#fff',
							}}
						>
							HDCP SCORE
						</h1>
					</Container>
					<Container
						display='flex'
						alignItems='center'
						justifyContent='center'
						flex={1}
					>
						<h1 style={{ color: '#fff' }}>{user.hdcp}</h1>
					</Container>
				</Container>
			</Container>
		</Container>
	);
}
