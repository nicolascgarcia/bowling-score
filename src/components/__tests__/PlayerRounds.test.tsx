/* eslint-disable import/no-extraneous-dependencies */
import { createRef } from 'react';
import renderer from 'react-test-renderer';
import {
	Score,
	ScoreType,
	Player,
} from '../../contexts/GameContext/gameContext.types';
import PlayerRounds from '../PlayerRounds';

describe('<PlayerRounds/>', () => {
	it('should render PlayerRounds', () => {
		const ref = createRef<HTMLDivElement>();
		const scores: Score[] = new Array(9).fill(null).map(() => ({
			firstShot: 0,
			secondShot: 0,
			score: 0,
			shots: 0,
			type: ScoreType.NORMAL,
		}));

		scores.push({
			firstShot: 0,
			secondShot: 0,
			thirdShot: 0,
			score: 0,
			shots: 0,
			type: ScoreType.NORMAL,
		});

		const player: Player = {
			name: 'Nicolas',
			photo: '',
			scores,
			currentRound: 0,
			ended: false,
			hdcp: 0,
		};

		const component = renderer.create(
			<PlayerRounds
				currentPlayer={0}
				user={player}
				playerRef={ref}
				userIndex={0}
			/>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
