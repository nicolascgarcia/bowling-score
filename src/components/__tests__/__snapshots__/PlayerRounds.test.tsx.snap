// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerRounds/> should render PlayerRounds 1`] = `
<div
  className="sc-bczRLJ dgolPA"
  display="flex"
>
  <div
    className="sc-bczRLJ gxVZDJ"
    display="flex"
    width={1}
  >
    <img
      alt="Nicolas"
      height={50}
      src=""
      width={50}
    />
    <div
      className="sc-bczRLJ kvQOPU"
    >
      <h1
        className="sc-gsnTZi gZIovQ"
      >
        Nicolas
      </h1>
    </div>
  </div>
  <div
    className="sc-bczRLJ iOidDR"
    display="flex"
  >
    <div
      className="sc-bczRLJ jkldZp"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          1
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score0"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score0"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score0"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          2
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score1"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score1"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score1"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          3
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score2"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score2"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score2"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          4
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score3"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score3"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score3"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          5
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score4"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score4"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score4"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          6
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score5"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score5"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score5"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          7
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score6"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score6"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score6"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          8
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score7"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score7"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score7"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          9
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score8"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score8"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score8"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ kDsxdZ"
      display="flex"
    >
      <div
        className="sc-bczRLJ jAFwIZ"
        display="flex"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          10
        </h1>
      </div>
      <div
        className="sc-bczRLJ gfvHoF"
        display="flex"
      >
        <div
          className="sc-bczRLJ heaOkQ"
          display="flex"
        >
          <div
            className="sc-bczRLJ eqsrV"
            width={0.5}
          >
            <h2
              aria-label="firstshot-user0-score9"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
          <div
            className="sc-bczRLJ fIFoXR"
            display="flex"
            width={0.5}
          >
            <h2
              aria-label="secondshot-user0-score9"
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
            <h2
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              0
            </h2>
          </div>
        </div>
        <div
          className="sc-bczRLJ dbmcqy"
          display="flex"
          width={1}
        >
          <h2
            aria-label="score-user0-score9"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            0
          </h2>
        </div>
      </div>
    </div>
    <div
      className="sc-bczRLJ bMJQTi"
      display="flex"
    >
      <div
        className="sc-bczRLJ dXJsed"
        width={1}
      >
        <h1
          style={
            Object {
              "color": "#fff",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          HDCP SCORE
        </h1>
      </div>
      <div
        className="sc-bczRLJ cPyNVh"
        display="flex"
      >
        <h1
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          0
        </h1>
      </div>
    </div>
  </div>
</div>
`;
